### DISCUSSION TITLE
Simple way to obtain Moon Angles in Kotlin.

### DISCUSSION THREAD

**Discussion by amoledwatchfaces** [cosinekitty/astronomy#298] [Q&A] [answered]
Hello, I'm looking for the best performance oriented way to obtain these statistics for the Moon based on observers location and time:

1. Illuminated portion of the moon in %
2. moon phase value (0->1 or 0 -> 360 or -180->+180)
3. Position angle of axis
4. Bright limb angle
5. Parallactic angle

These are used to draw a bitmap of the moon: [DrawMoonBitmap.kt](https://github.com/amoledwatchfaces/Complications-Suite-Wear-OS/blob/master/wear/src/main/java/com/weartools/weekdayutccomp/DrawMoonBitmap.kt)

I'm currently using [SunMoonCalculator](https://github.com/amoledwatchfaces/Complications-Suite-Wear-OS/blob/master/wear/src/main/java/com/weartools/weekdayutccomp/SunMoonCalculator.java) from http://conga.oan.es/~alonso/doku.php?id=blog:sun_moon_position
(It's just one java file with all computations)

Is it possible to obtain these values with Astronomy Engine in the quick and easy way?

Thanks for your answers!

Tomas

**Comment by cosinekitty** [comment, marked as answer]
Hello @amoledwatchfaces and welcome to the project!

Most of this will be easy in Astronomy Engine. You can start with the [illumination](https://github.com/cosinekitty/astronomy/blob/master/source/kotlin/doc/illumination.md) function. It returns an [Illumination](https://github.com/cosinekitty/astronomy/blob/master/source/kotlin/doc/-illumination-info/index.md) object, which tells you illumination phase angle (the angle in degrees between the Sun and the Earth, as seen from the body), and the illuminated phase fraction (from 0.0 to 1.0).

It is possible to calculate the Moon's axis using the function [rotationAxis](https://github.com/cosinekitty/astronomy/blob/master/source/kotlin/doc/rotation-axis.md). It gives you the direction of the axis in space as both a vector and as equatorial angles (right ascension and declination). Do you have a sense of what a "position angle" means in the context you need? That part may take some research.

About the bright limb angle and parallactic angle: I have had requests for these angles before, and generally they are both used as a means to find what the person really wants, which is the orientation of the lit side of the Moon as seen by an observer on the Earth, with respect to the observer's local horizon. In other words, you want to draw the way the Moon looks to a person standing at a given latitude and longitude at a given time.

I approached this problem myself without thinking of bright limb and parallactic angles. Instead, I calculate the angle of the sunlit side of the Moon from the observer's zenith direction. I have demo programs for this, but not in Kotlin yet. Probably the easiest one to understand will be the C# program [camera.cs](https://github.com/cosinekitty/astronomy/blob/master/demo/csharp/camera/camera.cs). The formulas in that code could be ported to Kotlin fairly easily, because the Astronomy Engine functions and types are similar between C# and Kotlin.

If your goal is to draw the Moon the way it appears to an observer, that camera.cs source code may help. Or maybe you are trying to do something else, in which case I can help you figure that out instead. So let's keep the conversation going and I'm sure I can help you in your goal.

**Reply by amoledwatchfaces** [reply]
Hi. Thanks for answering so quickly! Yes, that camera.cs program could be helpful. I'll look into it further. I'm trying to achieve exactly the same thing. I'm drawing moon image based on illuminated fraction (0-100%). Currently I'm rotating it based on these three angles, axis, bright limb angle (I'm not sure but I think it's an angle between body north and center of the bright limb), and parallactic angle (zenith to celestial north).

I'm trying to find some easier way to do it. One calculation, then just use of the angles. 

Thanks again for your answer 🙂

**Reply by cosinekitty** [reply]
You're welcome. Feel free to post follow-up questions here as you investigate more.
